refbox/src/refBox.js

```diff
@@ -223,6 +230,18 @@ class RefBox extends React.Component {
     });
   }
 
+  updateAvailableChallenges(challenges) {
+    this.setState({
+      availableChallenges: challenges
+    })
```

Comment:
Will this not delete all other keys in the state?

Reply:
Nope, check, e.g., [this post](https://www.freecodecamp.org/news/functional-setstate-is-the-future-of-react-374f30401b6b/):
'
Note how `setState()` works. You pass it an object containing part(s) of the state you want to update. In other words, the object you pass would have keys corresponding to the keys in the component state, then `setState()` updates or sets the state by merging the object to the state. Thus, “set-State”.
'